torchgeo/trainers/detection.py

```diff
@@ -206,18 +206,19 @@ def configure_models(self) -> None:
     def configure_metrics(self) -> None:
         """Initialize the performance metrics.
 
-        * Mean Average Precision (mAP): Computes the Mean-Average-Precision (mAP) and
-          Mean-Average-Recall (mAR) for object detection. Prediction is based on the
-          intersection over union (IoU) between the predicted bounding boxes and the
-          ground truth bounding boxes. Uses 'macro' averaging. Higher values are better.
+        * :class:`~torchmetrics.detection.mean_ap.MeanAveragePrecision`: Mean average
+          precision (mAP) and mean average recall (mAR). Precision is the number of
+          true positives divided by the number of true positives + false positives.
+          Recall is the number of true positives divived by the number of true positives
+          + false negatives. Uses 'macro' averaging. Higher values are better.
 
         .. note::
            * 'Micro' averaging suits overall performance evaluation but may not
              reflect minority class accuracy.
            * 'Macro' averaging gives equal weight to each class, and is useful for
              balanced performance assessment across imbalanced classes.
         """
-        metrics = MetricCollection([MeanAveragePrecision()])
+        metrics = MetricCollection([MeanAveragePrecision(average="macro")])
```

Comment:
This is the default, but it's better to be explicit about this since the default may change and we're documenting it due to confusion.

Reply:
I typically see mAP@50 in papers, we are reporting this? The torchmetrics docs state `map score is calculated with @[ IoU=self.iou_thresholds | area=all | max_dets=max_detection_thresholds ].` but I don't know what that value is by default

Reply:
I think we are reporting all of the metrics you displayed in #1923, although I don't know the details of what those all mean.

Reply:
Yes I'm wondering if its best to return them all and let the user choose which are significant, or return a documented subset. Probably the first is preferable, but adding more docs over time could be helpful

Reply:
I would much rather improve the torchmetrics docs than our docs.

Reply:
I agree, I think we should link to the torchmetrics docs instead of copy/pasting their definitions here which seems like just we are just adding boilerplate to our docs when users can go look at all the torchmetrics definitions.